同时安装多个版本optix时，add_packages中添加版本限制报错

### Xmake Version

xmake v2.9.5+master.654f14971

### Operating System Version and Architecture

windows 11

### Describe Bug

同时安装optix的7.3.0与8.0.0后，使用add_requires("optix 8.0.0")无法正确识别8.0.0版本。如果去掉版本限制只能识别7.3.0

### Expected Behavior

能够正确识别版本

### Project Configuration
```lua
add_rules("mode.debug", "mode.release")
set_project("ROTW-Optix-XMake")
set_version("0.0.1")
add_requires("cuda", "optix 8.0.0", "stb", "assimp")

target("rtow")
    set_languages("cxx17")
    add_packages("cuda", "optix", "stb", "assimp")
    set_kind("binary")
    add_files("*.cpp")
    add_cugencodes("native")
    add_includedirs("../include")
    add_defines(
        "OPTIX_IR_FILE=R\"($(scriptdir)/$(buildir)/optix-ir/optix-ir.optixir)\"",
        "OUTPUT_DIR=R\"($(scriptdir)/$(buildir)/)\"",
        "ASSETS_DIR=R\"($(scriptdir)/assets/)\""
    )
```
### Additional Information and Error Logs
```console
[0mchecking for cl.exe ... [38;2;0;255;0;1mC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe[0m
[0mchecking for Microsoft Visual Studio (x64) version ... [38;2;0;255;0;1m2019[0m
[0mchecking for Microsoft C/C++ Compiler (x64) version ... [38;2;0;255;0;1m19.29.30156[0m
[0m[38;2;255;255;0;1mcheckinfo: [0;2mcannot runv(zig.exe version), No such file or directory[0m
[0mchecking for zig ... [38;2;255;0;0;1mno[0m
[0m[38;2;255;255;0;1mcheckinfo: [0;2mcannot runv(zig.exe version), No such file or directory[0m
[0mchecking for zig ... [38;2;255;0;0;1mno[0m
[0m[38;2;255;255;0;1mcheckinfo: [0;2mcannot runv(nim.exe --version), No such file or directory[0m
[0mchecking for nim ... [38;2;255;0;0;1mno[0m
[0m[38;2;255;255;0;1mcheckinfo: [0;2mcannot runv(nim.exe --version), No such file or directory[0m
[0mchecking for nim ... [38;2;255;0;0;1mno[0m
[0mchecking for git ... [38;2;0;255;0;1mok[0m
[0m[38;2;255;255;0;1mcheckinfo: [0;2mcannot runv(gzip.exe --version), No such file or directory[0m
[0mchecking for gzip ... [38;2;255;0;0;1mno[0m
[0mchecking for 7z ... [38;2;0;255;0;1mC:\Users\aaa\xmake\winenv\bin\7z[0m
git rev-parse HEAD
[0mchecking for Cuda SDK directory ... [38;2;0;255;0;1mC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7[0m
finding cuda from xmake ..
[0mchecking for xmake::cuda ... [38;2;255;0;0;1mno[0m
[0mchecking for xmake-repo::cuda ... [38;2;0;255;0;1mcuda [0m
finding optix from xmake ..
[0mchecking for xmake::optix ... [38;2;255;0;0;1mno[0m
finding optix from vcpkg ..
finding optix from conan ..
finding optix from system ..
[0mchecking for cl.exe ... [38;2;0;255;0;1mC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe[0m
[0m[2mchecking for the c compiler (cc) ... [38;2;0;255;0;1mcl.exe[0m
[0m[2m> "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe" -c -nologo -FoC:\Users\aaa\AppData\Local\Temp\.xmake\241023\_0D0A0D72DB41496085C1167573A5B1E0.o C:\Users\aaa\AppData\Local\Temp\.xmake\241023\_775FB00633BC4F66BF87D39CCDE5CF30.c[0m
[0mchecking for cl.exe ... [38;2;0;255;0;1mC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe[0m
[0m[2mchecking for the c++ compiler (cxx) ... [38;2;0;255;0;1mcl.exe[0m
[0mchecking for link.exe ... [38;2;0;255;0;1mC:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\link.exe[0m
[0m[2mchecking for the linker (ld) ... [38;2;0;255;0;1mlink.exe[0m
[0m[2m> "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\link.exe" -nologo -dynamicbase -nxcompat -machine:x64 optix.lib -out:C:\Users\aaa\AppData\Local\Temp\.xmake\241023\_0D0A0D72DB41496085C1167573A5B1E0.b C:\Users\aaa\AppData\Local\Temp\.xmake\241023\_0D0A0D72DB41496085C1167573A5B1E0.o[0m
[0m[2m> checking for c links(optix)[0m
[0m[2m> checking for c snippet(find_package/optix)[0m
[0m[38;2;255;255;0;1mcheckinfo:[0;2m ...gramdir\core\sandbox\modules\import\core\tool\linker.lua:75: @programdir\modules\core\tools\link.lua:175: LINK : fatal error LNK1181: 无法打开输入文件“optix.lib”

stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\core\tools\link.lua:175]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:123]: in function 'try'
    [@programdir\modules\core\tools\link.lua:150]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:241]:
    [@programdir\core\tool\linker.lua:232]: in function 'link'
    [...gramdir\core\sandbox\modules\import\core\tool\linker.lua:73]: in function 'link'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:249]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:241]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\lib\detect\check_cxsnippets.lua:236]:
    [@programdir\modules\package\manager\system\find_package.lua:73]:
    [@programdir\modules\package\manager\find_package.lua:86]: in function '_find_package_with_builtin_rule'
    [@programdir\modules\package\manager\find_package.lua:132]: in function '_find_package'
    [@programdir\modules\package\manager\find_package.lua:196]:
    [@programdir\modules\lib\detect\find_package.lua:86]:
    [@programdir\core\package\package.lua:1902]: in function '_fetch_library'
    [@programdir\core\package\package.lua:2072]: in function 'fetch'
    [...modules\private\action\require\impl\install_packages.lua:340]: in function 'jobfunc'
    [@programdir\modules\async\runjobs.lua:241]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:241]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:117]: in function 'try'
    [@programdir\modules\async\runjobs.lua:223]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:
[0m
[0mchecking for optix ... [38;2;255;0;0;1mno[0m
finding stb from xmake ..
[0mchecking for xmake::stb ... [38;2;0;255;0;1mstb 2024.06.01[0m
[0mchecking for cmake ... [38;2;255;0;0;1mno[0m
[0mchecking for cmake ... [38;2;255;0;0;1mno[0m
[0mchecking for cmake ... [38;2;255;0;0;1mno[0m
[0mchecking for cmake ... [38;2;0;255;0;1mC:\Program Files\CMake\bin\cmake[0m
finding zlib from xmake ..
[0mchecking for xmake::zlib ... [38;2;0;255;0;1mzlib v1.3.1[0m
finding minizip from xmake ..
[0mchecking for xmake::minizip ... [38;2;0;255;0;1mminizip v1.2.13[0m
finding assimp from xmake ..
[0mchecking for xmake::assimp ... [38;2;0;255;0;1massimp v5.4.3[0m
[0m[1;38;2;255;0;0;1merror: [0m@programdir\core\main.lua:329: @programdir\core\sandbox\modules\import\core\base\task.lua:65: @programdir\modules\private\action\require\install.lua:53: The packages(optix) not found, please run `xmake repo -u` first!
stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:1004]:
    [@programdir\modules\private\action\require\install.lua:53]: in function '_check_missing_packages'
    [@programdir\modules\private\action\require\install.lua:87]:
    [@programdir\actions\config\main.lua:391]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:241]:
    [@programdir\core\base\task.lua:491]: in function 'run'
    [@programdir\core\sandbox\modules\import\core\base\task.lua:63]: in function 'run'
    [@programdir\actions\build\main.lua:191]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:241]:
    [@programdir\core\base\task.lua:491]: in function 'run'
    [@programdir\core\main.lua:327]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:406]:

stack traceback:
	[C]: in function 'error'
	@programdir\core\base\os.lua:1004: in function 'base/os.raiselevel'
	(...tail calls...)
	@programdir\core\main.lua:329: in upvalue 'cotask'
	@programdir\core\base\scheduler.lua:406: in function <@programdir\core\base\scheduler.lua:399>[0m
```

--- a/packages/o/optix/xmake.lua
+++ b/packages/o/optix/xmake.lua
@@ -6,11 +6,14 @@ package("optix")
     on_fetch(function (package, opt)
         if opt.system then
             import("lib.detect.find_path")
+            import("core.base.semver")
 
             local paths = {"$(env OptiX_ROOT)"}
             if package:is_plat("windows") then
                 for _, dir in ipairs(os.dirs("$(env PROGRAMDATA)/NVIDIA Corporation/OptiX SDK *.*.*")) do
-                    table.insert(paths, dir)
+                    if semver.satisfies(dir:match("OptiX SDK (%d+%.%d+%.%d+)"), package:version_str()) then
+                        table.insert(paths, dir)
+                    end
                 end
             end
 

--- a/packages/o/optix/xmake.lua
+++ b/packages/o/optix/xmake.lua
@@ -11,7 +11,7 @@ package("optix")
             local paths = {"$(env OptiX_ROOT)"}
             if package:is_plat("windows") then
                 for _, dir in ipairs(os.dirs("$(env PROGRAMDATA)/NVIDIA Corporation/OptiX SDK *.*.*")) do
-                    if semver.satisfies(dir:match("OptiX SDK (%d+%.%d+%.%d+)"), package:version_str()) then
+                    if package:version_str() == "latest" or semver.satisfies(dir:match("OptiX SDK (%d+%.%d+%.%d+)"), package:version_str()) then
                         table.insert(paths, dir)
                     end
                 end

